lib/iris/coords.py

```diff
@@ -1502,8 +1454,13 @@ def climatological(self):
         Always reads as False if there are no bounds.
         On set, the input value is cast to a boolean, exceptions raised
         if units are not time units or if there are no bounds.
+
         """
-        return self._climatological if self.has_bounds() else False
+        if not self.has_bounds():
+            self._metadata.climatological = False
+        if not self.units.is_time_reference():
+            self._metadata.climatological = False
+        return self._metadata.climatological
```

Comment:
I have come around to the idea of this. 
We do a `is_time_reference` check in the setter, so we are being careful with what is being set as having climatological bounds.
There is a risk is if someone modifies an already existing climatological coordinate to  something like a latitude coordinate, without updating the climatological property. In these cases, I would prefer we ensure we save things out correctly (i.e. pick up that this coordinate shouldn't be climatological). It would only be a problem if they were to access coord._metadata.climatological directly, but they shouldn't be doing that
.
So this just needs some testing

Reply:
@lbdreyer Thanks for taking the time to think about this one, much appreciated. 

It was a bit cheeky of me to include this in this PR, but since I was in the space, I thought I'd try to address something that reeked of a potential issue for users.

I'll add some tests :smile: